sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -57,6 +61,29 @@ trait DateTimeFormatterHelper {
     }
     formatter
   }
+
+  // When legacy time parser policy set to EXCEPTION, check whether we will get different results
+  // between legacy format and new format. For legacy parser, DateTimeParseException will not be
+  // thrown. On the contrary, if the legacy policy set to CORRECTED, DateTimeParseException will
+  // address by the caller side.
+  protected def checkDiffResult[T](
+    s: String, legacyParseFunc: String => T): PartialFunction[Throwable, T] = {
+    case e: DateTimeParseException if LegacyBehaviorPolicy.withName(
```

Comment:
seems it's worth to have a method in `SQLConf` to return the policy, as it will be used in many places.